wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/AuthorizingFileContentProvider.java

```diff
@@ -48,26 +52,41 @@ public AuthorizingFileContentProvider(
 
   @Override
   public String fetchContent(String fileURL) throws IOException, DevfileException {
-    return fetchContent(fileURL, false);
+    return fetchContent(fileURL, false, null);
+  }
+
+  @Override
+  public String fetchContent(String fileURL, String credentials)
+      throws IOException, DevfileException {
+    return fetchContent(fileURL, false, credentials);
   }
 
   @Override
   public String fetchContentWithoutAuthentication(String fileURL)
       throws IOException, DevfileException {
-    return fetchContent(fileURL, true);
+    return fetchContent(fileURL, true, null);
   }
 
-  protected String fetchContent(String fileURL, boolean skipAuthentication)
+  private String fetchContent(
+      String fileURL, boolean skipAuthentication, @Nullable String credentials)
```

Comment:
would be nice to use Optional here also

Reply:
I don't think that passing `Optional` as a parameter is a good idea: https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments
This is a private method so I don't think it will affect something.